models/repo.go

```diff
@@ -2684,11 +2720,13 @@ func (repo *Repository) UploadAvatar(data []byte) error {
 		return fmt.Errorf("UploadAvatar: Update repository avatar: %v", err)
 	}
 
-	if err := os.MkdirAll(setting.RepositoryAvatarUploadPath, os.ModePerm); err != nil {
-		return fmt.Errorf("UploadAvatar: Failed to create dir %s: %v", setting.RepositoryAvatarUploadPath, err)
+	fs := storage.FileStorage{
+		Ctx:      context.Background(),
+		Path:     setting.RepositoryAvatarUploadPath,
+		FileName: repo.Avatar,
```

Comment:
What happens if two users choose the same file name for their repo avatars?

Reply:
Here, `<repo.ID>-<md5.Sum(data)>` is used as  `repo.Avatar`. So even if two user uses same file,  repo.ID will differ them.
https://github.com/go-gitea/gitea/blob/98c0c8ea2e0d4896cc3cc00423f0d2952fcbad21/models/repo.go#L2718